File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/companies/CompanyService.java
package dev.reuss.tmdb.domain.companies;

import dev.reuss.tmdb.common.image.CompanyImages;
import dev.reuss.tmdb.domain.companies.model.Company;
import dev.reuss.tmdb.domain.companies.model.CompanyAlternativeNames;
import dev.reuss.tmdb.value.id.CompanyId;

/**
 * Service for loading TMDB company metadata.
 */
public interface CompanyService {

    /**
     * Loads company details by company id.
     *
     * @param companyId the company id
     * @return company details
     */
    Company details(CompanyId companyId);

    /**
     * Loads alternative names for a company.
     *
     * @param companyId the company id
     * @return alternative company names
     */
    CompanyAlternativeNames alternativeNames(CompanyId companyId);

    CompanyImages images(CompanyId companyId);

}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/companies/CompanyService.kt
package dev.reuss.tmdb.domain.companies

import dev.reuss.tmdb.common.image.CompanyImages
import dev.reuss.tmdb.domain.companies.model.Company
import dev.reuss.tmdb.domain.companies.model.CompanyAlternativeNames
import dev.reuss.tmdb.value.id.CompanyId

/** Loads TMDB production-company metadata. */
interface CompanyService {
    /** Returns the company identified by [companyId]. */
    fun details(companyId: CompanyId): Company

    /** Returns alternative names recorded for the company. */
    fun alternativeNames(companyId: CompanyId): CompanyAlternativeNames

    /** Returns the company's PNG and SVG logos. */
    fun images(companyId: CompanyId): CompanyImages
}
